src/laser_core/migration.py

```diff
@@ -0,0 +1,394 @@
+"""
+This module provides various functions to calculate migration networks based on different models,
+including the gravity model, competing destinations model, Stouffer's model, and the radiation model.
+Additionally, it includes a utility function to calculate the great-circle distance between two points
+on the Earth's surface using the Haversine formula.
+
+Functions:
+    gravity(pops: np.ndarray, distances: np.ndarray, k: float, a: float, b: float, c: float, max_frac: Union[float, None]=None, **kwargs) -> np.ndarray:
+
+    row_normalizer(network: np.ndarray, max_rowsum: float) -> np.ndarray:
+        Normalize the rows of a given network matrix such that no row sum exceeds a specified maximum value.
+
+    competing_destinations(pops: np.ndarray, distances: np.ndarray, b: float, c: float, delta: float, **params) -> np.ndarray:
+
+    stouffer(pops: np.ndarray, distances: np.ndarray, k: float, a: float, b: float, include_home: bool, **params) -> np.ndarray:
+        Compute a migration network using a modified Stouffer's model.
+
+    radiation(pops: np.ndarray, distances: np.ndarray, k: float, include_home: bool, **params) -> np.ndarray:
+
+    distance(lat1: float, lon1: float, lat2: float, lon2: float) -> float:
+        Calculate the great-circle distance between two points on the Earth's surface using the Haversine formula.
+"""
+
+from numbers import Number
+
+import numpy as np
+
+
+def shared_sanity_checks(pops, distances, **params):
```

Comment:
Let name this simply `_sanity_checks` - the leading underscore will indicate that it isn't intended for anyone outside this module.

Let's also move this down toward the end of the file.